src/poprox_storage/repositories/data_stores/db.py

```diff
@@ -16,10 +18,67 @@
 logger.setLevel(logging.DEBUG)
 
 
+DbRepositories: type[BaseModel] = create_model(
+    "DbRepositories",
+    conn=(Connection, None),
+    __config__=ConfigDict(arbitrary_types_allowed=True),
+)
+
+
+def camel_to_snake(s):
+    return "".join(["_" + c.lower() if c.isupper() else c for c in s]).lstrip("_")
+
+
+def db_repositories(*args):
+    if len(args) == 1 and not isinstance(args[0], list):
+        names = [args[0]]
+    else:
+        names = args
+
+    def inner_decorator(handler):
+        @wraps(handler)
+        def wrapper(event, context):
+            type_hints = get_type_hints(DbRepositories)
+            with DB_ENGINE.connect() as conn:
+                # create Repositories object and pre-construct all requested repository objects
+                repos = DbRepositories(conn=conn)
+                for name in names:
+                    repo_type = type_hints[name]
+                    setattr(repos, name, repo_type(conn))
+                return handler(event, context, repos)
```

Comment:
The magic here gets a bit simpler because we no longer have explicitly annotated `DbSomethingRepository | None` types, so we don't have to extract type arguments from union types